disperser/cmd/batcher/main.go

```diff
@@ -43,15 +51,43 @@ func main() {
 	app.Description = "Service for creating a batch from queued blobs, distributing coded chunks to nodes, and confirming onchain"
 
 	app.Action = RunBatcher
-	err := app.Run(os.Args)
-	if err != nil {
-		log.Fatalf("application failed: %v", err)
+	// Run the CLI app in a separate goroutine
+	go func() {
+		if err := app.Run(os.Args); err != nil {
+			errorChan <- err
+		}
+	}()
+
+	if _, err := os.Create(healthProbePath); err != nil {
+		log.Printf("Failed to create readiness file: %v", err)
+	}
+	// Start File based liveness probe
+	go heartBeatMonitor(healthProbePath, probeChan, maxStallDuration)
+
+	ticker := time.NewTicker(118 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			// Signal the health probe to update the file
+			probeChan <- time.Now()
+			log.Printf("File Probe Timer Ticked: time %v Sent\n", time.Now())
+		case err := <-errorChan:
+			log.Printf("Batcher exited with error: %v\n", err)
+			close(probeChan) // Signal the health probe to stop
+			return
+		}
 	}
-
-	select {}
 }
 
 func RunBatcher(ctx *cli.Context) error {
+
+	// Clean up readiness file
+	if err := os.Remove(readinessProbePath); err != nil && !os.IsNotExist(err) {
```

Comment:
why do we only report failure when `os.IsNotExist(err)`? shouldn't we report whenver `err != nil?`